### ISSUE TITLE
compilation error on debian:testing with gcc 6.1.1

### ISSUE THREAD

**Issue by davidsblom** [patflick/mxx#11]
A compilation error occurred when compiling with debian:testing using gcc 6.1.1. The latest version of ubuntu works just fine though.

Here is the log:

```
In file included from ../thirdParty/mxx/include/mxx/samplesort.hpp:31:0,
                 from ../thirdParty/mxx/include/mxx/sort.hpp:34,
                 from ElRBFMeshMotionSolver.C:7:
../thirdParty/mxx/include/mxx/bitonicsort.hpp: In function 'void mxx::impl::bitonic_merge(_Iterator, _Iterator, _Compare, const mxx::comm&, int, int, int)':
../thirdParty/mxx/include/mxx/bitonicsort.hpp:103:34: error: there are no arguments to 'log' that depend on a template parameter, so a declaration of 'log' must be available [-fpermissive]
     int p2 = pow(2, ceil(log(size)/log(2)));
                                  ^
../thirdParty/mxx/include/mxx/bitonicsort.hpp:103:34: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
../thirdParty/mxx/include/mxx/bitonicsort.hpp:103:41: error: there are no arguments to 'log' that depend on a template parameter, so a declaration of 'log' must be available [-fpermissive]
     int p2 = pow(2, ceil(log(size)/log(2)));
                                         ^
../thirdParty/mxx/include/mxx/bitonicsort.hpp: In function 'void mxx::impl::bitonic_sort_rec(_Iterator, _Iterator, _Compare, const mxx::comm&, int, int, bool)':
../thirdParty/mxx/include/mxx/bitonicsort.hpp:127:34: error: there are no arguments to 'log' that depend on a template parameter, so a declaration of 'log' must be available [-fpermissive]
     int p2 = pow(2, ceil(log(size)/log(2)));
                                  ^
../thirdParty/mxx/include/mxx/bitonicsort.hpp:127:41: error: there are no arguments to 'log' that depend on a template parameter, so a declaration of 'log' must be available [-fpermissive]
     int p2 = pow(2, ceil(log(size)/log(2)));
                                         ^
../thirdParty/mxx/include/mxx/bitonicsort.hpp: In instantiation of 'void mxx::impl::bitonic_sort_rec(_Iterator, _Iterator, _Compare, const mxx::comm&, int, int, bool) [with _Iterator = __gnu_cxx::__normal_iterator<std::tuple<int, double, double, double>*, std::vector<std::tuple<int, double, double, double> > >; _Compare = Foam::ElRBFMeshMotionSolver::solve()::<lambda(const tuple_type&, const tuple_type&)>]':
../thirdParty/mxx/include/mxx/bitonicsort.hpp:166:27:   required from 'void mxx::bitonic_sort(_Iterator, _Iterator, _Compare, const mxx::comm&) [with _Iterator = __gnu_cxx::__normal_iterator<std::tuple<int, double, double, double>*, std::vector<std::tuple<int, double, double, double> > >; _Compare = Foam::ElRBFMeshMotionSolver::solve()::<lambda(const tuple_type&, const tuple_type&)>]'
../thirdParty/mxx/include/mxx/samplesort.hpp:175:17:   required from 'std::vector<typename std::iterator_traits<_Iter>::value_type> mxx::impl::sample_block_decomp(_Iterator, _Iterator, _Compare, size_t, const mxx::comm&) [with _Iterator = __gnu_cxx::__normal_iterator<std::tuple<int, double, double, double>*, std::vector<std::tuple<int, double, double, double> > >; _Compare = Foam::ElRBFMeshMotionSolver::solve()::<lambda(const tuple_type&, const tuple_type&)>; typename std::iterator_traits<_Iter>::value_type = std::tuple<int, double, double, double>; size_t = long unsigned int]'
../thirdParty/mxx/include/mxx/samplesort.hpp:343:46:   required from 'void mxx::impl::samplesort(_Iterator, _Iterator, _Compare, const mxx::comm&) [with _Iterator = __gnu_cxx::__normal_iterator<std::tuple<int, double, double, double>*, std::vector<std::tuple<int, double, double, double> > >; _Compare = Foam::ElRBFMeshMotionSolver::solve()::<lambda(const tuple_type&, const tuple_type&)>; bool _Stable = false]'
../thirdParty/mxx/include/mxx/sort.hpp:41:49:   required from 'void mxx::sort(_Iterator, _Iterator, _Compare, const mxx::comm&) [with _Iterator = __gnu_cxx::__normal_iterator<std::tuple<int, double, double, double>*, std::vector<std::tuple<int, double, double, double> > >; _Compare = Foam::ElRBFMeshMotionSolver::solve()::<lambda(const tuple_type&, const tuple_type&)>]'
ElRBFMeshMotionSolver.C:231:66:   required from here
../thirdParty/mxx/include/mxx/bitonicsort.hpp:127:29: error: 'log' was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
     int p2 = pow(2, ceil(log(size)/log(2)));
```

**Comment by davidsblom** [comment]
I guess this can be fixed by including the appropriate header file at the top:

```
#include <tgmath.h>
```

**Comment by patflick** [comment]
Fixed in #12 . Thanks David.
